modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/RebalanceStatisticsUtils.java

```diff
@@ -79,536 +67,267 @@ private RebalanceStatisticsUtils() {
+            f = true;
 
-        Comparator<RebalanceFuture> startTimeCmp = comparingLong(fut -> fut.stat.startTime);
-        Comparator<RebalanceFuture> startTimeCmpReversed = startTimeCmp.reversed();
+            sb.a(p).a(" = ");
+            for (ClusterNode partNode : partNodes) {
+                if (!f)
+                    sb.a(',');
+                else
+                    f = false;
 
-        Comparator<PartitionStatistics> partIdCmp = comparingInt(value -> value.id);
-        Comparator<ClusterNode> nodeAliasesCmp = comparingInt(nodeAliases::get);
+                sb.a('[').a(affNodes.get(partNode)).a(aff.primaryPartitions(partNode.id()).contains(p) ? ",pr" : ",bu")
+                    .a(supStats.containsKey(partNode) ? ",su" : "").a("]");
 
-        for (Entry<CacheGroupContext, Collection<RebalanceFuture>> rebFutrsEntry : rebFutrs.entrySet()) {
-            CacheGroupContext cacheGrpCtx = rebFutrsEntry.getKey();
+            }
+            sb.a(!part.getValue() ? " h " : " ");
+        }
 
-            joiner.add("[id=" + cacheGrpCtx.groupId() + ",")
-                .add("name=" + cacheGrpCtx.cacheOrGroupName() + "]");
+        sb.a("Aliases: pr - primary, bu - backup, su - supplier node, h - historical, nodeId mapping ")
+            .a("(nodeId=id,consistentId) ");
 
-            List<RebalanceFuture> successFutures = new ArrayList<>();
+        partNodes.addAll(affNodes.keySet());
 
-            for (RebalanceFuture rebalanceFuture : rebFutrsEntry.getValue()) {
-                if (rebalanceFuture.isDone() && rebalanceFuture.get())
-                    successFutures.add(rebalanceFuture);
-            }
+        for (ClusterNode affNode : partNodes)
+            sb.a(supInfo(affNodes.get(affNode), affNode));
 
-            if (successFutures.isEmpty())
-                return;
+        return sb.toString();
+    }
 
-            successFutures.sort(startTimeCmpReversed);
+    /**
+     * Creating a string representation of total rebalance statistics for all
+     * cache groups for logging.
+     *
+     * @param totalStat Statistics of rebalance for cache groups.
+     * @return String representation of total rebalance statistics
+     *      for all cache groups.
+     */
+    public static String totalRebalanceStatistic(Map<CacheGroupContext, CacheGroupTotalRebalanceStatistics> totalStat) {
+        SB sb = new SB();
 
-            RebalanceFuture lastSuccessFuture = successFutures.get(0);
+        long start = totalStat.values().stream().mapToLong(CacheGroupTotalRebalanceStatistics::start).min().orElse(0);
+        long end = totalStat.values().stream().mapToLong(CacheGroupTotalRebalanceStatistics::end).max().orElse(0);
 
-            AffinityAssignment affinity = cacheGrpCtx.affinity().cachedAffinity(lastSuccessFuture.topologyVersion());
+        sb.a("Total information (including successful and not rebalances): [").a(time(start, end)).a("] ");
```

Comment:
Let's start this log message with word "rebalance".
It should be clear what's it about from the first glance.